misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperCodeActionUtil.java

```diff
@@ -75,11 +83,18 @@
     private static final int LEFT_SYMBOL_INDEX = 0;
     private static Cache<Integer, String> mappingCache =
             CacheBuilder.newBuilder().maximumSize(MAXIMUM_CACHE_SIZE).build();
+    private static HashMap<String, String> isOptionalMap = new HashMap<>();
```

Comment:
Why do we use static maps in a util? Is there a special reason for caching the older values and reuse them?

Reply:
I'm not caching older values. Just need to store them till generating the mapping function.